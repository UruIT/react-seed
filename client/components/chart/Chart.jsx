import React from 'react';
import style from './Chart.scss';
import PropTypes from 'prop-types';
import { Line } from 'react-chartjs-2';

const options = onClick => ({
	responsive: true,
	title: {
		display: true,
		text: 'POC Chart.js using react-chartjs-2'
	},
	tooltips: {
		// mode: 'point' would fill the tooltip with the information of the hovered point in that position
		mode: 'x',
		position: 'nearest',
		backgroundColor: '#f0efef',
		borderWidth: 2,
		borderColor: '#848484',
		titleFontSize: 24,
		titleFontStyle: 'bold',
		titleFontColor: '#b9b0b0',
		titleSpacing: 4,
		bodyFontSize: 16,
		bodyFontColor: '#848484',
		bodySpacing: 3,
		xPadding: 15,
		yPadding: 10,
		caretPadding: 10,
		caretSize: 10,
		callbacks: {
			label: function(tooltipItems) {
				return 'U$S ' + tooltipItems.yLabel.toLocaleString();
			}
		}
	},
	hover: {
		mode: 'point',
		onHover: function(evt, points) {
			if (evt.toElement.attributes.style) {
				if (points.length === 0) {
					evt.toElement.attributes.style.nodeValue = evt.toElement.attributes.style.nodeValue.replace(
						'cursor: pointer;',
						''
					);
					return;
				}
				var res = evt.toElement.attributes.style.nodeValue.match(/cursor: pointer;/);
				if (res == null) {
					evt.toElement.attributes.style.nodeValue += 'cursor: pointer;';
				}
			}
		}
	},
	scales: {
		yAxes: [
			{
				ticks: {
					callback: function(value) {
						return 'U$S ' + value.toLocaleString();
					}
				}
			}
		]
	},
	onClick: (event, elements) => {
		if (elements.length > 0) {
			onClick({
				elements
			});
		}
	}
});

const Chart = ({ data, onClick }) =>
	<div className={style.chart}>
		<Line data={data} options={options(onClick)} />
	</div>;

Chart.propTypes = {
	onClick: PropTypes.func,
	data: PropTypes.object.isRequired
};

Chart.defaultProps = {
	onClick: () => {}
};

export default Chart;
