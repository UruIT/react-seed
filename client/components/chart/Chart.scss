.chart {
	height: auto;
	margin: auto;
}
