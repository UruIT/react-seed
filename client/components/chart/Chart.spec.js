import React from 'react';
import Chart from './Chart';
import ShallowRenderer from 'react-test-renderer/shallow';

test('renders Chart', () => {
	const renderer = new ShallowRenderer();
	const data = {};
	const tree = renderer.render(<Chart data={data} />);

	expect(tree).toMatchSnapshot();
});
