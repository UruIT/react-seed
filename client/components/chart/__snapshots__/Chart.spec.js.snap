// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Chart 1`] = `
<div
  className="chart"
>
  <Line
    data={Object {}}
    options={
      Object {
        "hover": Object {
          "mode": "point",
          "onHover": [Function],
        },
        "onClick": [Function],
        "responsive": true,
        "scales": Object {
          "yAxes": Array [
            Object {
              "ticks": Object {
                "callback": [Function],
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "POC Chart.js using react-chartjs-2",
        },
        "tooltips": Object {
          "backgroundColor": "#f0efef",
          "bodyFontColor": "#848484",
          "bodyFontSize": 16,
          "bodySpacing": 3,
          "borderColor": "#848484",
          "borderWidth": 2,
          "callbacks": Object {
            "label": [Function],
          },
          "caretPadding": 10,
          "caretSize": 10,
          "mode": "x",
          "position": "nearest",
          "titleFontColor": "#b9b0b0",
          "titleFontSize": 24,
          "titleFontStyle": "bold",
          "titleSpacing": 4,
          "xPadding": 15,
          "yPadding": 10,
        },
      }
    }
  />
</div>
`;
